File: 0001-two-sum/0001-two-sum.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
      for (int i = 0; i < nums.length; i++) {
          for (int j = i + 1; j < nums.length; j++) {
              if ((nums[i] + nums[j]) == target) {
                  return new int[]{i, j};
              }
          }
      }
      return null;
  }
}

// Runtime: 45 ms (33.57%), Memory: 43.57 MB (99.84%)

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in hashmap:
                return [i, hashmap[complement]]
            hashmap[nums[i]] = i

        return []

# Runtime: 0 ms (100.00%), Memory: 20.29 MB (66.52%)
# Time complexity: O(n)
# Space complexity: O(n)
